backend/coreapp/asm_diff_wrapper.py

```diff
@@ -13,6 +13,14 @@
 
 MAX_FUNC_SIZE_LINES = 5000
 
+class DiffError(Exception):
+    def __init__(self, message: str):
+        super().__init__(f"Diff error: {message}")
+
+class ObjdumpError(Exception):
+    def __init__(self, message: str):
+        super().__init__(f"Objdump error: {message}")
+
```

Comment:
It might be helpful to define a base exception class here? Something like this?

```python
class SubprocessError(Exception):
    SUBPROCESS_NAME: ClassVar[str] = "Subprocess"
    stdout: str
    stderr: str

    def __init__(self, message: str):
        super().__init__(f"{self.SUBPROCESS_NAME} error: {message}")
        self.stdout = ""
        self.stderr = ""

    @staticmethod
    def from_process_error(self, ex: CalledProcessError) -> SubprocessError:
        error = SubprocessError(f"{ex.cmd[0]} returned {ex.returncode}")
        error.stdout = ex.stdout
        error.stderr = ex.stderr

class DiffError(SubprocessError):
    SUBPROCESS_NAME: ClassVar[str] = "Diff"

class ObjdumpError(SubprocessError):
    SUBPROCESS_NAME: ClassVar[str] = "Objdump"

# ...

try:
    pass
except subprocess.CalledProcessError as e:
    raise DiffError.from_process_error(e)
```

(You could tweak things to always include stderr/stdout in the message, etc.)